plenario/database.py

```diff
@@ -1,15 +1,81 @@
-from sqlalchemy import create_engine
+import os
+from sqlalchemy import create_engine, and_, text, func
 from sqlalchemy.orm import sessionmaker, scoped_session
-from sqlalchemy.pool import NullPool
 from sqlalchemy.ext.declarative import declarative_base
 
 from plenario.settings import DATABASE_CONN
 
-
-app_engine = create_engine(DATABASE_CONN, convert_unicode=True)
+if os.environ.get('WORKER'):
+    app_engine = create_engine(DATABASE_CONN, convert_unicode=True, pool_size=8, max_overflow=4, pool_timeout=60)
+else:
+    app_engine = create_engine(DATABASE_CONN, convert_unicode=True, pool_size=4, max_overflow=3, pool_timeout=60)
 
 session = scoped_session(sessionmaker(bind=app_engine,
                                       autocommit=False,
                                       autoflush=False, expire_on_commit=False))
 Base = declarative_base(bind=app_engine)
 Base.query = session.query_property()
+
+
+# Efficient query of large datasets (for use in DataDump)
+# Referenced from https://bitbucket.org/zzzeek/sqlalchemy/wiki/UsageRecipes/WindowedRangeQuery
+def column_windows(session, column, windowsize):
```

Comment:
This range query is used within datadump to make download requests for large datasets must faster